### PR TITLE
PXP-11248 PXP-11258 POST /auth/mapping token support

### PR DESCRIPTION
**Pull request by paulineribeyre**
Jira Tickets: [PXP-11248](https://ctds-planx.atlassian.net/browse/PXP-11248) and [PXP-11258](https://ctds-planx.atlassian.net/browse/PXP-11258)

Changes from https://github.com/uc-cdis/arborist/pull/159 + unit tests + fix `handleAuthMappingPOST` to work when no body is provided + minor refactoring

### New Features
- `POST /auth/mapping` supports parsing username and client ID from a token instead of from the request body.
- `POST /auth/mapping` supports tokens produced from a "client_credentials" OIDC flow.

### Deployment changes

[PXP-11248]: https://ctds-planx.atlassian.net/browse/PXP-11248?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[PXP-11258]: https://ctds-planx.atlassian.net/browse/PXP-11258?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: arborist/server_test.go [hunk 1]
```diff
@@ -3074,91 +3073,177 @@ func TestServer(t *testing.T) {
...
+					expectedMappings := make(arborist.AuthMapping)
+					for k, v := range anonymousAuthMapping {
+						expectedMappings[k] = v
+					}
+					for k, v := range loggedInAuthMapping {
+						expectedMappings[k] = v
+					}
+					msg := fmt.Sprintf("Expected response to be these auth mappings from anonymous and logged-in groups: %v", expectedMappings)
+					for resource, actions := range result {
+						assert.Contains(t, expectedMappings, resource, msg)
```
Comment (k-burt-uch) [inline arborist/server_test.go:3109]: Just for my understanding - if this "hulkhogan12" has auth mappings other than loggedIn or anonymous, this assert.Contains will fail. Is that correct?

Comment (paulineribeyre) [reply arborist/server_test.go:3109]: yes 👍

File: arborist/server.go [hunk 2]
```diff
@@ -102,7 +102,7 @@ func (server *Server) MakeRouter(out io.Writer) http.Handler {
 	router.HandleFunc("/health", server.handleHealth).Methods("GET")
 
 	router.Handle("/auth/mapping", http.HandlerFunc(server.handleAuthMappingGET)).Methods("GET")
-	router.Handle("/auth/mapping", http.HandlerFunc(server.parseJSON(server.handleAuthMappingPOST))).Methods("POST")
+	router.Handle("/auth/mapping", http.HandlerFunc(server.handleAuthMappingPOST)).Methods("POST")
```
Comment (AlbertSnows) [inline arborist/server.go:105]: Is the main purpose of this change to bypass the `baseHandler` function call?

Comment (paulineribeyre) [reply arborist/server.go:105]: The endpoint used to only accept input in the request body. Now that we allow passing a token and no body, we don't need `parseJSON` (it errors when there is no body). So i refactored the "body parsing" logic out of `parseJSON` so i could call it manually in `handleAuthMappingPOST()`, only when we need it. When only a token is passed, we don't call it

**Review by k-burt-uch — APPROVED** [review]

File: arborist/server.go (hunk 2 above)
Comment (AlbertSnows) [reply arborist/server.go:105]: that makes sense. so if this expression is satisfied 
`authHeader := r.Header.Get("Authorization"); authHeader != ""`
then we don't parse the body.

**Pull request merged**